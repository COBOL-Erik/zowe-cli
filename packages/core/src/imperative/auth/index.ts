/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./builders/AuthCommandBuilder";
export * from "./builders/AuthLoginCommandBuilder";
export * from "./builders/AuthLogoutCommandBuilder";
export * from "./builders/CompleteAuthGroupBuilder";

export * from "./doc/IAuthHandlerApi";

export * from "./handlers/AbstractAuthHandler";
export * from "./handlers/BaseAuthHandler";
