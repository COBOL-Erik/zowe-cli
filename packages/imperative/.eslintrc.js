module.exports = {
    ...require("../../.eslintrc.js"),
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module",
        "project": [
            "./tsconfig.json",
            "./__tests__/tsconfig.json",
            "./tsconfig-tests.json"
        ]
    }
}