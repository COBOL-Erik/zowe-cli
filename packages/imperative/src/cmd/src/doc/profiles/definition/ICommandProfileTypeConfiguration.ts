/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandProfileSchema } from "./ICommandProfileSchema";
import { IProfileTypeConfiguration } from "../../../../..";
import { ICommandProfileAuthConfig } from "./ICommandProfileAuthConfig";

/**
 * Cmd packages additions to the profile manager type configuration document. Used by the CliProfileManager. Allows
 * profiles to be built from command arguments. See the "CliProfileManager" for more information.
 * @export
 * @interface ICommandProfileTypeConfiguration
 * @extends {IProfileTypeConfiguration}
 */
export interface ICommandProfileTypeConfiguration extends IProfileTypeConfiguration {
    /**
     * The JSON schema document. The schema document provides a way to enforce the contents of a profile. The schema
     * conforms exactly to the JSON schema specification. You must supply all properties you would like validated
     * for correctness on the schema, except for "dependencies". Dependency schema checking is generated automatically
     * if you populate the dependencies property of this document.
     *
     * @type {ICommandProfileSchema}
     */
    schema: ICommandProfileSchema;

    /**
     * Configuration for authentication services to associate with this profile type.
     */
    authConfig?: ICommandProfileAuthConfig[];
}
