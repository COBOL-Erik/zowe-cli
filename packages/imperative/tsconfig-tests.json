{
  "compilerOptions": {
    "types": [
      "node",
      "jest"
    ],
    "lib": [
      "esnext"
    ],
    "experimentalDecorators": true,
    "target": "es2015",
    "module": "commonjs",
    "noEmit": true,
    "declaration": true,
    "moduleResolution": "node",
    "noImplicitAny": true,
    "outDir": "./lib",
    "preserveConstEnums": true,
    "removeComments": false,
    "pretty": true,
    "sourceMap": true,
    "newLine": "lf"
  },
  "typeRoots": [
    "./node_modules/@types"
  ],
  "include": [
    "**/__tests__/*",
    "**/__tests__/**/*",
    "**/__mocks__/*"
  ],
  "exclude": [
    "lib",
    "node_modules"
  ],
  "files": [
    "../../__types__/wontache.d.ts"
  ],
  "allowJs": true
}
