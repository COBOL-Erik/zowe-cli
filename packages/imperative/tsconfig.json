{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "stripInternal": true
  },
  "include": [
    "./src"
  ]
}