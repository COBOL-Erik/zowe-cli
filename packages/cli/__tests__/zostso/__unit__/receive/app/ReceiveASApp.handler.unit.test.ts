/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import { ZosmfRestClient } from "@zowe/core-for-zowe-sdk";
import { ReceiveTsoApp } from "@zowe/zos-tso-for-zowe-sdk";
import * as ReceiveASAppHandler from "../../../../../src/zostso/receive/app/ReceiveASApp.handler";
import * as ReceiveASAppDefinition from "../../../../../src/zostso/receive/app/ReceiveASApp.definition";
import { IHandlerParameters } from "@zowe/imperative";
import { mockHandlerParameters } from "@zowe/cli-test-utils";
import { UNIT_TEST_ZOSMF_PROF_OPTS } from "../../../../../../../__tests__/__src__/TestConstants";

const DEFAULT_PARAMETERS: IHandlerParameters = mockHandlerParameters({
    arguments: UNIT_TEST_ZOSMF_PROF_OPTS,
    positionals: ["zos-tso","r","app"],
    definition: ReceiveASAppDefinition.ReceiveASApp
});
const MOCK_RECEIVE_RESPONSE: any = {
    version: undefined,
    reused: false,
    timeout: false,
    servletKey: "JR897694-122-aabyaaaj",
    queueID: null,
    tsoData: [
        {
            "TSO MESSAGE": { VERSION: "0100", DATA: "Processing started." },
        },
        {
            "TSO MESSAGE": { VERSION: "0100", DATA: "READY" },
        },
    ],
};

describe("receive TSO app handler behavior", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should properly receive and parse data from receive TSO response", async () => {
        jest.spyOn(ZosmfRestClient, "getExpectJSON").mockResolvedValueOnce(
            MOCK_RECEIVE_RESPONSE
        );
        const receiveSpy = jest.spyOn(ReceiveTsoApp,"receive");
        const handler = new ReceiveASAppHandler.default();
        const params = Object.assign({}, ...[DEFAULT_PARAMETERS]);
        let error = undefined;
        params.arguments = {...params.arguments,account: "izuacct", appKey: "test2", servletKey: "JR897694-122-aabyaaaj", runUntilReady: true};
        try{
            await handler.process(params);
        }
        catch(e)
        {
            error = e;
        }
        expect(error).toBeUndefined();
        expect(receiveSpy).toHaveBeenCalledTimes(1);
    });
});
