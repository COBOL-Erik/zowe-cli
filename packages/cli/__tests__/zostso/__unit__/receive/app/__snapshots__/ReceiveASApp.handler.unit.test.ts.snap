// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receive TSO app handler behavior should properly receive and parse data from receive TSO response 1`] = `
"
"
`;

exports[`receive TSO app handler behavior should properly receive and parse data from receive TSO response 2`] = `"Processing started."`;

exports[`receive TSO app handler behavior should properly receive and parse data from receive TSO response 3`] = `"READY"`;

exports[`receive TSO app handler behavior should properly receive and parse data from receive TSO response 4`] = `
Object {
  "queueID": null,
  "reused": false,
  "servletKey": "ZOWEUSER-122-aabyaaaj",
  "timeout": false,
  "tsoData": Array [
    Object {
      "DATA": "Processing started.",
      "VERSION": "0100",
    },
    Object {
      "DATA": "READY",
      "VERSION": "0100",
    },
  ],
  "version": undefined,
}
`;
